stock_inventory/models/stock_quant.py

```diff
@@ -22,39 +23,44 @@ def _apply_inventory(self):
         res = super()._apply_inventory()
         record_moves = self.env["stock.move.line"]
         adjustment = self.env["stock.inventory"].browse()
-        for rec in self:
-            adjustment = rec.current_inventory_id
-            moves = record_moves.search(
-                [
-                    ("product_id", "=", rec.product_id.id),
-                    ("lot_id", "=", rec.lot_id.id),
-                    "|",
-                    ("location_id", "=", rec.location_id.id),
-                    ("location_dest_id", "=", rec.location_id.id),
-                ],
-                order="create_date asc",
-            ).filtered(
-                lambda x, rec=rec: not x.company_id.id
-                or not rec.company_id.id
-                or rec.company_id.id == x.company_id.id
-            )
-            if len(moves) == 0:
-                raise ValueError(_("No move lines have been created"))
-            move = moves[len(moves) - 1]
-            adjustment.stock_move_ids |= move
-            reference = move.reference
-            if adjustment.name and move.reference:
-                reference = adjustment.name + ": " + move.reference
-            elif adjustment.name:
-                reference = adjustment.name
-            move.write(
-                {
-                    "inventory_adjustment_id": adjustment.id,
-                    "reference": reference,
-                }
-            )
-            rec.to_do = False
-            rec.current_inventory_id = False
+
+        for adjustment, quant_list in groupby(
+            self, key=lambda x: x.current_inventory_id
+        ):
+            new_moves = self.env["stock.move.line"]
```

Comment:
This is more correct syntaxically:

```suggestion
            new_moves = self.env["stock.move.line"].browse()
```